### DISCUSSION TITLE
Spoolman, KAMP and Screws Tilt Adjust for KE

### DISCUSSION THREAD

**Discussion by SPYKE357** [Guilouz/Creality-Helper-Script-Wiki#565] [Features Request]
hi 
Please add Spoolman, KAMP and Screws Tilt Adjust for KE to the helper script. would be awesome

**Comment by Guilouz** [comment]
Spoolman is not compatible.

I don’t have KE, so just basic settings they no need specific configuration

**Reply by SPYKE357** [reply]
I know kamp and screws tilt adjust would be very popular with the KE users, i can give you the screws coordinates for the KE if that would help. i've been able to get both working on mine, but had to do a workaround using winscp and editing some lines. Please see if you can add this functionality as an optional install with the script as it would just make things so much easier for those of us that have modded our beds spacers. It would be highly appreciated. Kamp is always a nice to have.

**Reply by Guilouz** [reply]
Yes you can send me. Is it for ender 3 KE ?

**Reply by SPYKE357** [reply]
Yes it's for the Ender 3 KE

i have :

[screws_tilt_adjust]
screw1: 27,0.5
screw1_name: front left screw 
screw2: 196.6,0.5 
screw2_name: front right screw 
screw3: 196.6,169.6 
screw3_name: rear right screw 
screw4: 27,169.6 
screw4_name: rear left screw 
horizontal_move_z: 5 
speed: 150 
screw_thread: CW-M4


although i did recalibrate probe offset, which is 

x_offset: 1.1
y_offset: 27.2

stock offset is 

x_offset: 0
y_offset: 27

**Reply by foresterr** [reply]
One of those Ender 3 v3 KE users with whom this would be very popular reporting here!

I actually just measured my bed screw positions without messing with stock probe offset (in the vain hope it will just work ™) and came up with:

```
screw1: 27.5, -0.5
screw1_name: front left screw
screw2: 196.5, -0.5
screw2_name: front right screw
screw3: 27.5, 169.5
screw3_name: rear right screw
screw4: 196.5, 169.5
screw4_name: rear left screw
```
which is pretty close to what the OP provided.

Probably x1=27, x2=196, y1=0, y2=169 would be a good default for anyone to tweak from - it will still certainly be within screw area (screw socket diameter is ~10mm) and I don't think submillimeter accuracy would matter that greatly here.

**Comment by simonsays10** [comment]
> Yes it's for the Ender 3 KE
> 
> i have :
> 
> [screws_tilt_adjust] screw1: 27,0.5 screw1_name: front left screw screw2: 196.6,0.5 screw2_name: front right screw screw3: 196.6,169.6 screw3_name: rear right screw screw4: 27,169.6 screw4_name: rear left screw horizontal_move_z: 5 speed: 150 screw_thread: CW-M4
> 
> although i did recalibrate probe offset, which is
> 
> x_offset: 1.1 y_offset: 27.2
> 
> stock offset is
> 
> x_offset: 0 y_offset: 27

I tried to add this to my printer.cfg file and it gave me an error message and I had to remove it before I could use printer again.  Very interesting how you were able to get this to work on yours.  I have a KE also.   Following this post to see what happens.

**Reply by SPYKE357** [reply]
hi simonsays, add me on reddit Capable_Bed_5188, i have a temporary fix, it's not ideal but it'll get you the functionality, dm me as im not sure Guilouz would be happy with me discussing workarounds and backdooring into his scripts. The printer definitely has the filing structure for KAMP and Screws tilt adjust. It needs to be written specifically for the KE im assuming. It would definitely help to have kamp and screws tilt adjust as so many of us have done bed spacer mods.

**Comment by foresterr** [comment]
I opened a PR with screws tilt adjust support for KE, hopefully it's OK and can be used.

**Reply by SPYKE357** [reply]
awesome! i saw the Helper Script Update during the night with your pr added. Thank you to you and Guilouz. That was quick. Well done

**Comment by X-Ryl669** [comment]
What makes spoolman not compatible ?
